components/dada-parse/src/parameter_parser.rs

```diff
@@ -3,7 +3,7 @@ use crate::parser::Parser;
 use dada_ir::{class::Class, function::Function, parameter::Parameter, token_tree::TokenTree};
 use salsa::DebugWithDb;
 
-#[salsa::component(in crate::Jar ref)]
+#[salsa::component(jar = crate::Jar, return_ref)]
```

Comment:
What's `salsa::component` for? can't find it in salsa's document.  Can we use `salsa::memoized` here?

Reply:
Good question -- this has been renamed to the `specify` option for `salsa::tracked`, see [this part of the overview](https://salsa-rs.github.io/salsa/overview.html#specified-the-result-of-tracked-functions-for-particular-structs) for details.